src/core/qgsproject.h

```diff
@@ -575,6 +576,54 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      */
     QMap<QString, QgsMapLayer *> mapLayers() const;
 
+    /**
+     * Unzip a project
+     * \param filename The zip file to unzip
+     * \returns true if unzip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool unzip( const QString &filename );
+
+    /**
+     * Unzip a project with the current zip filename
+     * \returns true if unzip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool unzip();
+
+    /**
+     * Zip the project
+     * \param filename The zip filename
+     * \returns true if zip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool zip( const QString &filename );
```

Comment:
I am wondering if we need all these new methods in the public API - would it be possible to completely hide the details of zipping/unzipping inside QgsProject implementation? I would expect just one extra boolean "zipped" flag (with getter and setter): reading it would tell me if the project was zipped, setting it would determine whether the project should be zipped when written.

```python
project = QgsProject()
project.read('/home/xyz/project.qgz')
project.isZipped()  # returns True
```